spec/lib/rex/post/smb/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -13,6 +16,9 @@
   end
 
   before(:each) do
+    allow(client).to receive(:dispatcher).and_return(dispatcher)
+    allow(dispatcher).to receive(:tcp_socket).and_return(socket)
```

Comment:
```suggestion
```

```
  let(:simple_client) { instance_double(Rex::Proto::SMB::SimpleClient, dispatcher: dispatcher) }
  let(:dispatcher) { instance_double(RubySMB::Dispatcher::Socket, socket: socket) }
```

Not tested

Reply:
Was this cromulent in the end? 👀

Reply:
yup, sorry missed this somehow, works just fine